api/tests/src/search/backend/test_load_opportunities_to_index.py

```diff
@@ -151,3 +200,41 @@ def test_load_opportunities_to_index_index_does_not_exist(self, db_session, sear
 
         with pytest.raises(RuntimeError, match="please run the full refresh job"):
             load_opportunities_to_index.run()
+
+    def test_new_opportunity_gets_indexed(
+        self, db_session, test_opportunity, load_opportunities_to_index
+    ):
+        """Test that a new opportunity in the queue gets indexed"""
+        # Add to queue
+        OpportunitySearchIndexQueueFactory.create(
+            opportunity_id=test_opportunity.opportunity_id, has_update=True
+        )
+        db_session.commit()
```

Comment:
You can be slightly less verbose with just:
```suggestion
        OpportunitySearchIndexQueueFactory.create(opportunity=test_opportunity)
```
has_update has a default, and the opportunity/opportunity_id all get set together anyways (both at the factory and ORM level).

The factory `create` function also commits after it runs